### PR TITLE
Consolidate CI workflows on GH Actions

### PR DESCRIPTION
**Pull request by mroderick**
This PR consolidates the CI setup on GitHub Actions.

We've added the `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` as repository secrets, which are only available to contributors and not in forks.

We've tested the SauceLabs integration by changing the condition to this branch, i.e. `if: ${{ github.ref == 'refs/heads/consolideate-ci-setup' }}`, which showed that it is now working.

#### After merging

Once the branch has been merged, we need to remove the project in CircleCI and remove the configuration file here. Otherwise, CircleCI will block the branch from merging. Additionally, in the "Integrations" settings for the repository, CircleCI can be removed

### CODE REVIEW

**Review by hexeberlin — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -87,9 +87,48 @@ jobs:
...
+      - name: Upload coverage report
+        run: bash <(curl -s https://codecov.io/bash) -F unit -s coverage/lcov.info
+
+  saucelabs-test:
+    if: ${{ github.ref == 'refs/heads/master' }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [16]
```
Comment (mantoni) [inline .github/workflows/main.yml:106]: Why is this using a matrix with a single version?

**Review by mantoni — APPROVED** [review]
Just one minor comment. Thank you for this 🙌

File: .github/workflows/main.yml (hunk 1 above)
Comment (mroderick) [reply .github/workflows/main.yml:106]: Because lazy. I'll tidy it up

**Comment by mroderick** [comment]
@hexeberlin and I have moved all the repositories to use GitHub Actions and have retired CircleCI entirely

**Pull request merged**